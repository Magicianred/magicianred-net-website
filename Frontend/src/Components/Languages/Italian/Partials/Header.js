import React from 'react';
import PropTypes from 'prop-types';
import Nav from 'react-bootstrap/Nav';
import Navbar from 'react-bootstrap/Navbar';
import logo from './../../../../logo.svg';

import LanguagesSwitch from '../../../Partials/LanguagesSwitch';

const publicUrl = process.env.PUBLIC_URL;

/**
 * Functional Component for showing the header of the site. 
 * @component
 * @param {string} title Title of the site
 * @example
 * const sitename = 'my site name'
 * return (
 *   <Header title={sitename} />
 * )
 */
const Header = ({ title }) => {
    return (
        <header className="partial header" style={{ marginBottom: "15px" }}>
            <Navbar className="menu">
                <Navbar.Brand href={`${publicUrl}/#/`}>
                    {/* <img
                        src={logo}
                        width="30"
                        height="30"
                        className="d-inline-block align-top"
                        alt={`${title} logo`}
                    /> */}
                    </Navbar.Brand>
                    <Navbar.Toggle aria-controls="basic-navbar-nav" />
                    <Navbar.Collapse id="basic-navbar-nav">
                        <Nav className="mr-auto">
                            <Nav.Link className="header-link" id="home_link" href={`${publicUrl}/#/it/`}>Home</Nav.Link>
                            <Nav.Link className="header-link" id="web-project_link" href={`${publicUrl}/#/it/progetti-web`}>Progetti Web</Nav.Link>
                            <Nav.Link className="header-link" id="writer-project_link" href={`${publicUrl}/#/it/progetti-scrittura`}>Progetti Scrittura</Nav.Link>
                            <Nav.Link className="header-link" id="permacultura_link" href={`${publicUrl}/#/it/permacultura`}>Permacultura</Nav.Link>
                            <Nav.Link className="header-link" id="about_link" href={`${publicUrl}/#/it/about`}>About</Nav.Link>
                        </Nav>
                        <LanguagesSwitch />
                    </Navbar.Collapse>
            </Navbar>
            <section className="header-description">
                <div id="plandesign"><img src="/images/plans.png" alt="" /></div>
                <h1><a href="#">La Magia delle Idee <small><em>by Magicianred</em></small></a></h1>
                <p>spazio personale per dare libero sfogo alle idee. Non ci saranno solo idee concrete e pienamente sviluppate,
                ma germogli, semi, bozze e quant'altro mi verrà in mente. <br/>
                Molte saranno abbandonate, alcune vedranno la luce e si svilupperanno,
                molte altre invece saranno lasciate libere di circolare con licenze adatte per farle raccoglie a chi se la sentirà.</p>
            </section>
        </header>
    );
}

Header.propTypes = {
    /**
     * Title of the site
     */
    title: PropTypes.string.isRequired,
}

Header.defaultProps = {
    title: 'Default Site title'
}

export default Header;