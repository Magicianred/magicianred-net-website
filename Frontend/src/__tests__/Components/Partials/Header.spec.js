import React from 'react';
import { create } from 'react-test-renderer';

import Header from './../../../Components/Partials/Header'

describe("Show the Header site", () => {

    describe("Snapshots", () => {
    
        it("Matches the snapshot (with default title)", () => {
            const instance = create(<Header />);
            expect(instance.toJSON()).toMatchSnapshot();
        });
    
        it("Matches the snapshot (with title: 'My App Title')", () => {
            const instance = create(<Header title="My App Title" />);
            expect(instance.toJSON()).toMatchSnapshot();
        });
    
    });

    xit("Show Header site", () => {

    });
});