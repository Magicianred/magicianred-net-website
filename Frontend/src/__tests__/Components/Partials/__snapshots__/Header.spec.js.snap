// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Show the Header site Snapshots Matches the snapshot (with default title) 1`] = `
<header
  className="partial header"
>
  <nav
    className="navbar navbar-expand navbar-light bg-light"
  >
    <a
      className="navbar-brand"
      href="/"
    >
      <img
        alt="Default Site title logo"
        className="d-inline-block align-top"
        height="30"
        src="logo.svg"
        width="30"
      />
    </a>
    <button
      aria-controls="basic-navbar-nav"
      aria-label="Toggle navigation"
      className="navbar-toggler collapsed"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      aria-expanded={null}
      className="navbar-collapse collapse"
      id="basic-navbar-nav"
    >
      <div
        className="mr-auto navbar-nav"
        onKeyDown={[Function]}
      >
        <a
          className="nav-link"
          data-rb-event-key="/"
          href="/"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          Home
        </a>
        <a
          className="nav-link"
          data-rb-event-key="/about"
          href="/about"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          About Us
        </a>
        <a
          className="nav-link"
          data-rb-event-key="/contact"
          href="/contact"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          Contact Us
        </a>
        <a
          className="nav-link"
          data-rb-event-key="/abcdefg"
          href="/abcdefg"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          Not Found
        </a>
        <a
          className="nav-link"
          data-rb-event-key="/error"
          href="/error"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          Error Page
        </a>
        <a
          className="nav-link"
          data-rb-event-key="/docs"
          href="/docs"
          onClick={[Function]}
          onKeyDown={[Function]}
          target="_blank"
        >
          Docs [open new page]
        </a>
      </div>
    </div>
  </nav>
</header>
`;

exports[`Show the Header site Snapshots Matches the snapshot (with title: 'My App Title') 1`] = `
<header
  className="partial header"
>
  <nav
    className="navbar navbar-expand navbar-light bg-light"
  >
    <a
      className="navbar-brand"
      href="/"
    >
      <img
        alt="My App Title logo"
        className="d-inline-block align-top"
        height="30"
        src="logo.svg"
        width="30"
      />
    </a>
    <button
      aria-controls="basic-navbar-nav"
      aria-label="Toggle navigation"
      className="navbar-toggler collapsed"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      aria-expanded={null}
      className="navbar-collapse collapse"
      id="basic-navbar-nav"
    >
      <div
        className="mr-auto navbar-nav"
        onKeyDown={[Function]}
      >
        <a
          className="nav-link"
          data-rb-event-key="/"
          href="/"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          Home
        </a>
        <a
          className="nav-link"
          data-rb-event-key="/about"
          href="/about"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          About Us
        </a>
        <a
          className="nav-link"
          data-rb-event-key="/contact"
          href="/contact"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          Contact Us
        </a>
        <a
          className="nav-link"
          data-rb-event-key="/abcdefg"
          href="/abcdefg"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          Not Found
        </a>
        <a
          className="nav-link"
          data-rb-event-key="/error"
          href="/error"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          Error Page
        </a>
        <a
          className="nav-link"
          data-rb-event-key="/docs"
          href="/docs"
          onClick={[Function]}
          onKeyDown={[Function]}
          target="_blank"
        >
          Docs [open new page]
        </a>
      </div>
    </div>
  </nav>
</header>
`;
